### PR TITLE
Fix Godeps

### PR DESCRIPTION
**Pull request by soltysh**
It looks like we can't perform clean `hack/godep-restore.sh` run. I've tried to identified the problems and fix them as part of PR.

@ncdc this will be helpful for the rebase
@ironcladlou you've pinged me for having problems with it, see if those fixes help you
@deads2k || @liggitt ptal
@dcbw I've dropped test files from `github.com/containernetworking/cni` mind taking a look as well, if that doesn't bite you?
@mfojtik fyi

Generally, I'd like us to run additional test that verifies `hack/godep-restore.sh` works cleanly.

### CODE REVIEW

**Comment by soltysh** [comment]
[test]

**Comment by soltysh** [comment]
Flake #10988

**Comment by soltysh** [comment]
The last commit adds running `hack/godep-restore.sh` as last step in `make check`. I didn't want to put it in `verify` b/c that's one being run in travis and it's doubtful it'll fit in given time-limit.

**Comment by soltysh** [comment]
@smarterclayton fyi

**Comment by soltysh** [comment]
Same flake #10988, not going to re-run it more.

File: Makefile [hunk 1]
```diff
@@ -281,4 +282,4 @@ release-rpms: clean build-rpms
 #   make vendor-console
 vendor-console:
 	GIT_REF=$(GIT_REF) CONSOLE_REPO_PATH=$(CONSOLE_REPO_PATH) hack/vendor-console.sh
-.PHONY: vendor-console
```
Comment (mfojtik) [inline Makefile]: why this change?

Comment (soltysh) [inline Makefile]: Because my editor auto-magically adds new lines at the end of each file :dancer:

**Comment by soltysh** [comment]
This was picked up in #11957.

**Comment by soltysh** [comment]
@ncdc I've closed this, but it looks like prematurely. I wanted to submit a PR with just the last commit adding `hack/godep-restore.sh` to `make verify` but I see that script still fails with quite a few errors. Were you able to have a clean run with it?

**Comment by ncdc** [comment]
I realized on Friday that we still need some of your fixes
